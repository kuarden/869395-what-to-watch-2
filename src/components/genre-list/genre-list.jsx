import React, {PureComponent} from "react";
import PropTypes from 'prop-types';
import withActiveItem from "../../hocs/with-active-item";

class GenreList extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {genres, onGenreClick, activeItem = -1, onChangeActiveItem = () => ({})} = this.props;

    return <ul className="catalog__genres-list">
      { genres.map((genre, i) =>
        (<li className={`catalog__genres-item ${activeItem === i ? `catalog__genres-item--active` : ``}`}
          key={`catalog__genres-item--${genre}-${i}`}
          onClick={(event)=>{
            event.preventDefault();
            onChangeActiveItem(i);
            onGenreClick(genre);
          }}>
          <a href="#" className="catalog__genres-link">
            {genre}
          </a>
        </li>))}
    </ul>;
  }
}

GenreList.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.string),
  onGenreClick: PropTypes.func.isRequired,
  activeItem: PropTypes.number,
  onChangeActiveItem: PropTypes.func,
};

export default withActiveItem(GenreList);
