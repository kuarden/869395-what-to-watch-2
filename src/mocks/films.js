export const films = [
  {
    id: 1,
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Fantasy`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 2,
    name: `Bohemian Rhapsody`,
    image: `img/bohemian-rhapsody.jpg`,
    genre: `Drama`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 3,
    name: `Macbeth`,
    image: `img/macbeth.jpg`,
    genre: `Fantasy`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 4,
    name: `Aviator`,
    image: `img/aviator.jpg`,
    genre: `Adventures`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 5,
    name: `We need to talk about Kevin`,
    image: `img/we-need-to-talk-about-kevin.jpg`,
    genre: `Adventures`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 6,
    name: `What We Do in the Shadows`,
    image: `img/what-we-do-in-the-shadows.jpg`,
    genre: `Fantasy`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 7,
    name: `Revenant`,
    image: `img/revenant.jpg`,
    genre: `Drama`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 8,
    name: `Johnny English`,
    image: `img/johnny-english.jpg`,
    genre: `Comedy`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 9,
    name: `Shutter Island`,
    image: `img/shutter-island.jpg`,
    genre: `Thriller`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 10,
    name: `Pulp Fiction`,
    image: `img/pulp-fiction.jpg`,
    genre: `Action`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 11,
    name: `No Country for Old Men`,
    image: `img/no-country-for-old-men.jpg`,
    genre: `Action`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 12,
    name: `Snatch`,
    image: `img/snatch.jpg`,
    genre: `Action`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 13,
    name: `Moonrise Kingdom`,
    image: `img/moonrise-kingdom.jpg`,
    genre: `Fantasy`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 14,
    name: `Seven Years in Tibet`,
    image: `img/seven-years-in-tibet.jpg`,
    genre: `Adventures`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 15,
    name: `Midnigth Special`,
    image: `img/midnight-special.jpg`,
    genre: `Fantasy`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 16,
    name: `War of the Worlds`,
    image: `img/war-of-the-worlds.jpg`,
    genre: `Fantasy`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 17,
    name: `Dardjeeling Limited`,
    image: `img/dardjeeling-limited.jpg`,
    genre: `Adventures`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 18,
    name: `Orlando`,
    image: `img/orlando.jpg`,
    genre: `Adventures`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 19,
    name: `Mindhunter`,
    image: `img/mindhunter.jpg`,
    genre: `Action`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 20,
    name: `Midnight Special`,
    image: `img/midnight-special.jpg`,
    genre: `Fantasy`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
];
