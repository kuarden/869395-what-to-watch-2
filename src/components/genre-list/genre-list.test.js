import React from "react";
import Enzyme, {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import {GenresList} from "./genre-list.jsx";
import {films} from "../../mocks/films.js";

Enzyme.configure({adapter: new Adapter()});

it(`GenresList renders correctly`, () => {
  const tree = shallow(<GenresList
    films={films}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
