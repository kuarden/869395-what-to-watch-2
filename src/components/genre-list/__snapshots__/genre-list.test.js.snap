// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenresList renders correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <Connect(GenreItem)
    activeTab={false}
    key="All genres"
    name="All genres"
  />
  <Connect(GenreItem)
    activeTab={false}
    key="Fantasy"
    name="Fantasy"
  />
  <Connect(GenreItem)
    activeTab={false}
    key="Drama"
    name="Drama"
  />
  <Connect(GenreItem)
    activeTab={false}
    key="Adventures"
    name="Adventures"
  />
  <Connect(GenreItem)
    activeTab={false}
    key="Comedy"
    name="Comedy"
  />
  <Connect(GenreItem)
    activeTab={false}
    key="Thriller"
    name="Thriller"
  />
  <Connect(GenreItem)
    activeTab={false}
    key="Action"
    name="Action"
  />
</ul>
`;
